### PR TITLE
adds OAuth2Authenticator

### PR DESCRIPTION
**Pull request by ppelgrims**
I added an OAuth2Authenticator, with tests. I also added a bit of example configuration.
This is my first pull request, any feedback would be appreciated.

### CODE REVIEW

File: config/bitbucket.php [hunk 1]
```diff
@@ -34,7 +34,7 @@
     | Here are each of the connections setup for your application. Example
     | configuration has been included, but you may add as many connections as
     | you would like. Note that the two supported authentication methods are:
-    | "basic", and "token".
+    | "basic", "token" and "oath2".
```
Comment (GrahamCampbell) [inline config/bitbucket.php]: `"basic", "token", "oath", and "oath2".` :)

File: README.md [hunk 2]
```diff
@@ -54,7 +54,7 @@ This option (`'default'`) is where you may specify which of the connections belo
 
 ##### Bitbucket Connections
 
-This option (`'connections'`) is where each of the connections are setup for your application. Example configuration has been included, but you may add as many connections as you would like. Note that the supported authentication methods are: `"basic"` and `"token"`, `"oauth"`.
+This option (`'connections'`) is where each of the connections are setup for your application. Example configuration has been included, but you may add as many connections as you would like. Note that the supported authentication methods are: `"basic"`, `"token"`, `"oauth"` and `"oauth2"`.
```
Comment (GrahamCampbell) [inline README.md]: Please also use the Oxford comma in lists. :)

**Comment by GrahamCampbell** [comment]
Thanks for the PR. Very nice work. Just two minor comments, then it looks good to go. :)

Also, can I quickly confirm you've tested this works IRL, as well as having the unit tests?

**Comment by ppelgrims** [comment]
Yes, I tested this IRL, as well as having the unit tests.

**Pull request merged**